src/filters/__tests__/waterShed.test.ts

```diff
@@ -0,0 +1,169 @@
+import { computeThreshold } from '../..';
+import { createGreyImage } from '../../../test/testUtils';
+import { waterShed } from '../waterShed';
+
+describe('Test WaterShed Roi generation', () => {
+  it('basic test 1, without parameters/options', () => {
+    const image = createGreyImage([
+      [3, 3, 3, 3, 3],
+      [3, 2, 2, 2, 3],
+      [3, 2, 1, 2, 3],
+      [3, 2, 2, 2, 3],
+      [3, 3, 3, 3, 3],
+    ]);
+    const roiMapManager = waterShed(image, {});
+    const resultArray = new Int16Array([
+      -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+      -1, -1, -1, -1, -1, -1, -1,
+    ]);
+    expect(roiMapManager).toEqual({
+      map: {
+        data: resultArray,
+        nbPositive: 0,
+        nbNegative: 1,
+        width: 5,
+        height: 5,
+      },
+      whiteRois: [],
+      blackRois: [],
+    });
+  });
+
+  it('waterShed for a grey image', () => {
+    const image = createGreyImage([
+      [3, 3, 3, 3, 3, 3, 3, 3, 4, 4],
+      [3, 3, 2, 2, 2, 3, 3, 3, 4, 4],
+      [4, 3, 2, 1, 2, 2, 3, 3, 4, 4],
+      [4, 3, 2, 2, 2, 2, 3, 3, 3, 4],
+      [4, 4, 4, 3, 2, 3, 2, 3, 3, 4],
+      [4, 4, 4, 3, 3, 3, 3, 1, 3, 3],
+      [4, 3, 3, 3, 3, 3, 2, 2, 2, 3],
+      [4, 4, 3, 3, 3, 3, 2, 2, 2, 2],
+      [4, 4, 4, 4, 3, 2, 2, 2, 2, 3],
+      [4, 4, 4, 4, 3, 3, 3, 3, 2, 3],
+    ]);
+
+    const roiMapManager = waterShed(image, { threshold: 2 / 255 });
+
+    const resultArray = new Int16Array([
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -2, -2, -2, 0, 0, 0, 0, 0, 0, 0, -2,
+      -2, -2, -2, 0, 0, 0, 0, 0, 0, -2, -2, -2, -2, 0, 0, 0, 0, 0, 0, 0, 0, -2,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1, 0, 0, 0, 0, 0, 0, 0, 0, -1, -1,
+      -1, 0, 0, 0, 0, 0, 0, 0, -1, -1, -1, -1, 0, 0, 0, 0, 0, -1, -1, -1, -1, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, -1, 0,
+    ]);
```

Comment:
We need to find a better way because this is not really readable. I don't know if we have a helper somewhere (I didn't find one).

@targos Is it ok to add a method in 'testUtils' that allows to write

```js
getInt16Array(`
  0, 0, 1, 2
  3, 4, 5, 6,
  7, 8    9,  10
`) and returns a Int16Array ?
```

So something like:
```js
function getInt16Array(string) {
   return new Int16Array(string.split(/[, \r\n]+/.filter(value=>value).map(string=>Number(string)))
}
```

Reply:
We already have helpers to create images, so yeah sure (if it works the same way, or even better if it can reuse the code)